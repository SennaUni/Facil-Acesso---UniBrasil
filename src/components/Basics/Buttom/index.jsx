import React from 'react';
import { ActivityIndicator } from 'react-native';

import { Container, Title } from './styles';

export function Buttom({ title, loading = false, ...rest }) {
  return (
    <>
      { loading ? (
        <Container {...rest} loading={true} disabled={true}>
          <ActivityIndicator
            size='small'
            color='#FFF'
          />
          <Title>Loading</Title>
        </Container>
      ) : (
        <Container {...rest} >
          <Title>{title}</Title>
        </Container>
      )}
    </>
  );
}