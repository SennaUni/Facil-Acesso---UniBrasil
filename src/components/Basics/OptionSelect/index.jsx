import React from 'react';

import { FontAwesome } from '@expo/vector-icons';

import { Container, Picture, Label } from './styles';

export function OptionSelect({ item, selectedValue, onPress }) {

  return (
    <Container 
      backGround={item.id === selectedValue?.id ? '#EEE' : '#FFF'}
      onPress={onPress}
    >
      <Picture 
        source={{
          uri: item.avatarUrl,
        }}
      />
      <Label 
        fontWeight={item.id === selectedValue?.id ? 'bold' : 'normal'} 
      >
        {item.name}
      </Label>
      {/* { item.id === selectedValue?.id && (
        <FontAwesome
        name='check-square'
        size={22}
        color={'green'}
      />
      )} */}
    </Container>
  );
}
