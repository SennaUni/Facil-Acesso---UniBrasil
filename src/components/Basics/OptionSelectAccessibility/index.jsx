import React from 'react';

import { FontAwesome } from '@expo/vector-icons';

import { Container, Picture, Label } from './styles';

export function OptionSelectAccessibility({ item, selectedValue, onPress }) {

  return (
    <Container 
      backGround={item.id === selectedValue?.id ? '#8241B8' : '#FFF'}
      onPress={onPress}
    >
      <Picture 
        source={{
          uri: item.avatarUrl,
        }}
      />
      <Label 
        color={item.id === selectedValue?.id ? '#FFF' : '#555'}
      >
        {item.name}
      </Label>
      {/* { item.id === selectedValue?.id && (
        <FontAwesome
        name='check-square'
        size={22}
        color={'green'}
      />
      )} */}
    </Container>
  );
}
