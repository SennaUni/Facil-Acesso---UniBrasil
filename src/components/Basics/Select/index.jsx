import React, { useState, useRef, useEffect } from 'react';

import { TouchableOpacity, Modal, FlatList, SafeAreaView } from 'react-native';

import { FontAwesome } from '@expo/vector-icons';

import { useField } from '@unform/core';

import { Label, Container, ComboBox, ModalHeader, SelectTitle, ModalTitle, ModalClose } from './styles';

export function Select({ name, options, text, label, OptionComponent }) {
  const selectRef = useRef(null)

  // const { fieldName, registerField } = useField('teste');

  const [openModal, setOpenModal] = useState(false);
  const [selectedValue, setSelectedValue] = useState();

  function renderOption(item) {
    return (
      <OptionComponent 
        item={item}
        selectedValue={selectedValue}
        onPress={() => {
          setSelectedValue(item);
          setOpenModal(false);
        }}
      />
    )
  }

  // useEffect(() => {
  //   registerField({
  //     name: fieldName,
  //     ref: selectRef.current,
  //     path: 'value',
  //     getValue: ref => {
  //       return ref.value || ''
  //     },
  //     setValue: (ref, value) => {
  //       ref.setNativeProps({ text: value });
  //       ref.value = value
  //     },
  //     clearValue: ref => {
  //       ref.value = ''
  //     },
  //   })
  // }, [fieldName, registerField])

  return (
    <>
      <Label>
        {label}
      </Label>
      <Container >
        <ComboBox
          onPress={() => setOpenModal(true)}
        >
          <SelectTitle 
            numberOfLines={1}
          >
            {selectedValue?.name || text}
          </SelectTitle>
          <FontAwesome
            name='angle-down'
            size={26}
            color={'#000'}
          />
        </ComboBox>
        <Modal 
          animationType='slide'
          visible={openModal}
          onRequestClose={() => setOpenModal(false)}
        >
          <SafeAreaView>
            <ModalHeader>
              <TouchableOpacity
                onPress={() => setOpenModal(false)}
              >
                <FontAwesome
                  name='angle-down'
                  size={30}
                  color={'#000'}
                />
              </TouchableOpacity>
              <ModalTitle>
                {text}
              </ModalTitle>
              <TouchableOpacity
                onPress={() => setOpenModal(false)}
              >
                <ModalClose>
                  Cancelar
                </ModalClose>
              </TouchableOpacity>
            </ModalHeader>
            <FlatList 
              data={options ?? []}
              keyExtractor={item => String(item.id)}
              renderItem={({ item }) => renderOption(item) }
            />
          </SafeAreaView>
        </Modal>
      </Container>
    </>
  );
}