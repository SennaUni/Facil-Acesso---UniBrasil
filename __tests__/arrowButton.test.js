import { render, fireEvent } from '@testing-library/react-native';

import { ArrowButtom } from '../src/components/Basics/ArrowButtom';

test('Testando renderização com Loading False', async () => {
  const { toJSON } = render(
    <ArrowButtom 
      loading={false}
      gradient={[ '#A88BEB', '#8241B8' ]}
    />
  );

  expect(toJSON()).toMatchSnapshot();
});

test('Testando renderização com Loading True', async () => {
  const { toJSON } = render(
    <ArrowButtom 
      loading={true}
      gradient={[ '#A88BEB', '#8241B8' ]}
    />
  );

  expect(toJSON()).toMatchSnapshot();
});

test('Testando renderização com Loading False e Reverse True', async () => {
  const { toJSON } = render(
    <ArrowButtom 
      loading={false}
      gradient={[ '#A88BEB', '#8241B8' ]}
      reverse={true}
    />
  );

  expect(toJSON()).toMatchSnapshot();
});

test('Testando renderização com Loading True e Reverse True', async () => {
  const { toJSON } = render(
    <ArrowButtom 
      loading={true}
      gradient={[ '#A88BEB', '#8241B8' ]}
      reverse={true}
    />
  );

  expect(toJSON()).toMatchSnapshot();
});

