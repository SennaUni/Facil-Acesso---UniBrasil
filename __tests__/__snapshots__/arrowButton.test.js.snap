// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testando renderização com Loading False 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderRadius": 50,
      "height": 60,
      "opacity": 1,
      "overflow": "hidden",
      "width": 60,
    }
  }
>
  <BVLinearGradient
    colors={
      [
        4289235947,
        4286726584,
      ]
    }
    endPoint={
      {
        "x": 0.5,
        "y": 1,
      }
    }
    loading={false}
    locations={null}
    startPoint={
      {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      [
        {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        },
      ]
    }
  >
    <Text />
  </BVLinearGradient>
</View>
`;

exports[`Testando renderização com Loading False e Reverse True 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderRadius": 50,
      "height": 60,
      "opacity": 1,
      "overflow": "hidden",
      "width": 60,
    }
  }
>
  <BVLinearGradient
    colors={
      [
        4289235947,
        4286726584,
      ]
    }
    endPoint={
      {
        "x": 0.5,
        "y": 1,
      }
    }
    loading={false}
    locations={null}
    startPoint={
      {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      [
        {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        },
      ]
    }
  >
    <Text />
  </BVLinearGradient>
</View>
`;

exports[`Testando renderização com Loading True 1`] = `
<View
  accessibilityState={
    {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderRadius": 50,
      "height": 60,
      "opacity": 1,
      "overflow": "hidden",
      "width": 60,
    }
  }
>
  <BVLinearGradient
    colors={
      [
        4289235947,
        4286726584,
      ]
    }
    endPoint={
      {
        "x": 0.5,
        "y": 1,
      }
    }
    loading={true}
    locations={null}
    startPoint={
      {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      [
        {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "opacity": 0.7,
          "width": "100%",
        },
      ]
    }
  >
    <ActivityIndicator
      color="#FFF"
      size="small"
    />
  </BVLinearGradient>
</View>
`;

exports[`Testando renderização com Loading True e Reverse True 1`] = `
<View
  accessibilityState={
    {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderRadius": 50,
      "height": 60,
      "opacity": 1,
      "overflow": "hidden",
      "width": 60,
    }
  }
>
  <BVLinearGradient
    colors={
      [
        4289235947,
        4286726584,
      ]
    }
    endPoint={
      {
        "x": 0.5,
        "y": 1,
      }
    }
    loading={true}
    locations={null}
    startPoint={
      {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      [
        {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "opacity": 0.7,
          "width": "100%",
        },
      ]
    }
  >
    <ActivityIndicator
      color="#FFF"
      size="small"
    />
  </BVLinearGradient>
</View>
`;
